core/src/epicli/cli/engine/schema/SchemaValidator.py

```diff
@@ -61,8 +62,10 @@ def run(self):
             if hasattr(schema['properties']["specification"], '$ref'):
                 if schema['properties']["specification"]['$ref'] == '#/definitions/unvalidated_specification':
                     self.logger.warn('No specification validation for ' + doc.kind)
+            schema_dic = objdict_to_dict(schema)
```

Comment:
Could you explain the purpose of ObjDict and this operation with the same `src_class` and `dst_class` values?

```python
def objdict_to_dict(something):
    """Recursively convert any ObjDict to python dictionary."""
    return _nested_dict_to_dict(something, src_class=dict, dst_class=dict)
```

Reply:
ObjDict is just a helper class with some extension methods to query a dictionary like so:

```python
dict.value.nested_value.etc ....
```

Instead of like this with a normal dictionary :

```python
dict['value']['nested_value']['etc'] ...
```

Mostly used to make some of the operations in the infrastructure builders a bit more readable:

![image](https://user-images.githubusercontent.com/11056665/127449322-a6a85cc3-b59e-44fb-8ae9-0c4ae317014c.png)

We use objdict_to_dict and dict_to_objdict function to switch between them as some external libraries for loading and validating dont work with ObjDict`s but only with plain python Dict`s

As for the src_class and dst_class values, that is just from a helper method to indicate what to convert to what, ObjDict->Dict or Dict ->ObjDict

dict_to_objdict will use: 

```python
return _nested_dict_to_dict(something, src_class=dict, dst_class=ObjDict)
```